olp-cpp-sdk-dataservice-read/src/repositories/DataRepository.cpp

```diff
@@ -93,20 +94,6 @@ DataResponse DataRepository::GetVersionedData(
   }
 
   if (!request.GetDataHandle()) {
-    if (!request.GetVersion()) {
-      // Get latest version of the layer if it wasn't set by the user
-      CatalogVersionRequest version_request;
-      version_request.WithFetchOption(request.GetFetchOption())
-          .WithBillingTag(request.GetBillingTag());
-      auto latest_version_response =
-          repository::CatalogRepository::GetLatestVersion(
-              catalog, context, std::move(version_request), settings);
-      if (!latest_version_response.IsSuccessful()) {
-        return latest_version_response.GetError();
-      }
-      request.WithVersion(latest_version_response.GetResult().GetVersion());
-    }
```

Comment:
DataRequest::GetVersion is deprecated, but I think we can't just remove this block.
GetPartitionById() used below expecting for a version in the request, so this change breaking DataRepository::GetVersionedData() logic.
Is this related to merging requests?

Reply:
Actually we don't enter this branch. But I reverted this code for now.